server_reporter_custom_labels.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2016 Michal Witkowski. All Rights Reserved.
+// See LICENSE for licensing terms.
+
+package grpc_prometheus
+
+import (
+	"time"
+
+	"google.golang.org/grpc/codes"
+	"context"
+)
+
+type serverReporterCustomLabels struct {
+	metrics     *ServerMetricsCustomLabels
+	rpcType     grpcType
+	serviceName string
+	methodName  string
+	startTime   time.Time
+	labelValues []string
+}
+
+func newServerReporterCustomLabels(ctx context.Context, m *ServerMetricsCustomLabels, rpcType grpcType, fullMethod string) *serverReporterCustomLabels {
+	r := &serverReporterCustomLabels{
+		metrics: m,
+		rpcType: rpcType,
+	}
+
+	if r.metrics.serverHandledHistogramEnabled {
+		r.startTime = time.Now()
+	}
+
+	r.serviceName, r.methodName = splitMethodName(fullMethod)
+	labelValues := make([]string, len(m.customLabels), len(m.customLabels))
+	copy(labelValues, []string{string(r.rpcType), r.serviceName, r.methodName})
+	customLabelValues := m.customLabelsProvider.LabelsFromContext(ctx)
+	for i := len(defaultLables); i < len(m.customLabels); i++ {
+		si := i - len(defaultLables) //shifted index
```

Comment:
customLabels include default and additional. Do you just want to append the LabelsFromContext to that list?